kv/kvserver: TestFlowControlRaftSnapshotV2 failed

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/d1526788-0651-4582-b52a-0c0c23d50dd2?testReportRun=1&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [a1b013e763abd1454603985996a45663b6e6bcad](https://github.com/cockroachdb/cockroach/commits/a1b013e763abd1454603985996a45663b6e6bcad):


```
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_span_refresher.go:162 kvcoord.(*txnSpanRefresher).SendLocked ???
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_committer.go:188 kvcoord.(*txnCommitter).SendLocked ???
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_pipeliner.go:319 kvcoord.(*txnPipeliner).SendLocked ???
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_seq_num_allocator.go:111 kvcoord.(*txnSeqNumAllocator).SendLocked ???
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_heartbeater.go:265 kvcoord.(*txnHeartbeater).SendLocked ???
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_coord_sender.go:533 kvcoord.(*TxnCoordSender).Send ???
github.com/cockroachdb/cockroach/pkg/kv/db.go:1133 kv.(*DB).sendUsingSender ???
github.com/cockroachdb/cockroach/pkg/kv/txn.go:1287 kv.(*Txn).Send ???
github.com/cockroachdb/cockroach/pkg/kv/db.go:964 kv.sendAndFill ???
github.com/cockroachdb/cockroach/pkg/kv/txn.go:802 kv.(*Txn).Run ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_command.go:2588 kvserver.execChangeReplicasTxn.func2 ???
github.com/cockroachdb/cockroach/pkg/kv/txn.go:1051 kv.(*Txn).exec ???
github.com/cockroachdb/cockroach/pkg/kv/db.go:1097 kv.runTxn ???
github.com/cockroachdb/cockroach/pkg/kv/db.go:1060 kv.(*DB).TxnWithAdmissionControl ???
github.com/cockroachdb/cockroach/pkg/kv/db.go:1035 kv.(*DB).Txn ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_command.go:2452 kvserver.execChangeReplicasTxn ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_command.go:2081 kvserver.(*Replica).execReplicationChangesForVoters ???

goroutine 5125447 lock 0xc006f98468
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_create_replica.go:105 kvserver.(*Store).tryGetReplica ??? <<<<<
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_create_replica.go:104 kvserver.(*Store).tryGetReplica ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_create_replica.go:167 kvserver.(*Store).tryGetOrCreateReplica ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_create_replica.go:73 kvserver.(*Store).getOrCreateReplica ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:356 kvserver.(*Store).withReplicaForRequest ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:628 kvserver.(*Store).processRequestQueue ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:397 kvserver.(*raftSchedulerShard).worker ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:319 kvserver.(*raftScheduler).Start.func2 ???
github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:498 stop.(*Stopper).RunAsyncTaskEx.func2 ???

goroutine 5125447 lock 0xc006f98a68
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:2296 kvserver.(*Replica).withRaftGroup ??? <<<<<
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica.go:181 kvserver.(*ReplicaMutex).Lock ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:2295 kvserver.(*Replica).withRaftGroup ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_raft.go:610 kvserver.(*Replica).stepRaftGroupRaftMuLocked ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:406 kvserver.(*Store).processRaftRequestWithReplica ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:626 kvserver.(*Store).processRequestQueue.func1 ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:361 kvserver.(*Store).withReplicaForRequest ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:628 kvserver.(*Store).processRequestQueue ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:397 kvserver.(*raftSchedulerShard).worker ???
github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:319 kvserver.(*raftScheduler).Start.func2 ???
github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:498 stop.(*Stopper).RunAsyncTaskEx.func2 ???

goroutine 56 lock 0xc00f51a198
github.com/cockroachdb/cockroach/pkg/util/admission/grant_coordinator.go:749 admission.(*GrantCoordinator).CPULoad ??? <<<<<
github.com/cockroachdb/cockroach/pkg/util/admission/grant_coordinator.go:748 admission.(*GrantCoordinator).CPULoad ???
github.com/cockroachdb/cockroach/pkg/util/goschedstats/runnable.go:231 goschedstats.(*schedStatsTicker).getStatsOnTick ???
github.com/cockroachdb/cockroach/pkg/util/goschedstats/runnable.go:161 goschedstats.init.0.func1 ???



```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-43198

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/99137d9a-1226-4fa5-b3d8-23d4fa39bd13?testReportRun=1&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [a1b013e763abd1454603985996a45663b6e6bcad](https://github.com/cockroachdb/cockroach/commits/a1b013e763abd1454603985996a45663b6e6bcad):


```
        -  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.eval.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B     
        -  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B     
        +  kvflowcontrol.tokens.eval.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.eval.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
        +  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.returned                        | 4.0 KiB  
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B      
         
         
         -- Observe the total tracked tokens per-stream on n1. 2MiB is tracked for n1-n5;
         -- see last comment for an explanation why we're still deducting for n2, n3.
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
        --- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (23.83s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17325297?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17325297?buildTab=artifacts#/) on master @ [e49b56cc0e983ca2ec9a80ea898dbba60a1c6992](https://github.com/cockroachdb/cockroach/commits/e49b56cc0e983ca2ec9a80ea898dbba60a1c6992):


```
        -  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.eval.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B     
        -  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B     
        +  kvflowcontrol.tokens.eval.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.eval.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
        +  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.returned                        | 4.0 KiB  
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B      
         
         
         -- Observe the total tracked tokens per-stream on n1. 2MiB is tracked for n1-n5;
         -- see last comment for an explanation why we're still deducting for n2, n3.
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
        --- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (4.28s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2
    --- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2 (14.28s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/698b40b9-5776-4189-8fce-316febf5f82c?testReportRun=1&testReportShard=17&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [6d3f108b4de3ae24ebb1543ca3882144678f8fa2](https://github.com/cockroachdb/cockroach/commits/6d3f108b4de3ae24ebb1543ca3882144678f8fa2):


```
        -  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.eval.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B     
        -  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B     
        +  kvflowcontrol.tokens.eval.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.eval.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
        +  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.returned                        | 4.0 KiB  
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B      
         
         
         -- Observe the total tracked tokens per-stream on n1. 2MiB is tracked for n1-n5;
         -- see last comment for an explanation why we're still deducting for n2, n3.
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
        --- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (24.05s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=17</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/dc3aafa2-ee22-4cbd-87b7-2a28c689d87d?testReportRun=1&testReportShard=17&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [3eb2bb04abaaaaac92f3f5f2f6952a30ada78de5](https://github.com/cockroachdb/cockroach/commits/3eb2bb04abaaaaac92f3f5f2f6952a30ada78de5):


```
        -  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.eval.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B     
        -  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B     
        +  kvflowcontrol.tokens.eval.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.eval.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
        +  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.returned                        | 8.0 KiB  
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B      
         
         
         -- Observe the total tracked tokens per-stream on n1. 2MiB is tracked for n1-n5;
         -- see last comment for an explanation why we're still deducting for n2, n3.
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
        --- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (57.91s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=17</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17369038?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17369038?buildTab=artifacts#/) on master @ [b37ca45ae2eee82de92b778f4553f9fe8a19603e](https://github.com/cockroachdb/cockroach/commits/b37ca45ae2eee82de92b778f4553f9fe8a19603e):


```
        -  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.eval.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B     
        -  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B     
        +  kvflowcontrol.tokens.eval.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.eval.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
        +  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.returned                        | 4.0 KiB  
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B      
         
         
         -- Observe the total tracked tokens per-stream on n1. 2MiB is tracked for n1-n5;
         -- see last comment for an explanation why we're still deducting for n2, n3.
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
        --- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (8.60s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17371395?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17371395?buildTab=artifacts#/) on release-24.3 @ [4cbedefd790c75cb0f21f77ed8d917c8528a7d15](https://github.com/cockroachdb/cockroach/commits/4cbedefd790c75cb0f21f77ed8d917c8528a7d15):


```
        -  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.eval.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B     
        -  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B     
        +  kvflowcontrol.tokens.eval.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.eval.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
        +  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.returned                        | 4.0 KiB  
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B      
         
         
         -- Observe the total tracked tokens per-stream on n1. 2MiB is tracked for n1-n5;
         -- see last comment for an explanation why we're still deducting for n2, n3.
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
        --- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (4.96s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/7cafe118-d052-4671-8963-4873d7504c6e?testReportRun=2&testReportShard=18&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [472ea07a5232c98536293d13bb46cca59f9f2cd0](https://github.com/cockroachdb/cockroach/commits/472ea07a5232c98536293d13bb46cca59f9f2cd0):


```
        -  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.eval.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B     
        -  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B     
        +  kvflowcontrol.tokens.eval.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.eval.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
        +  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.returned                        | 4.0 KiB  
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B      
         
         
         -- Observe the total tracked tokens per-stream on n1. 2MiB is tracked for n1-n5;
         -- see last comment for an explanation why we're still deducting for n2, n3.
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
        --- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (24.48s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=18</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/092f05b7-b7cf-41a8-8de5-3bb3005e52e8?testReportRun=1&testReportShard=17&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-24.3 @ [4cbedefd790c75cb0f21f77ed8d917c8528a7d15](https://github.com/cockroachdb/cockroach/commits/4cbedefd790c75cb0f21f77ed8d917c8528a7d15):


```
        -  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.eval.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B     
        -  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B     
        +  kvflowcontrol.tokens.eval.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.eval.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
        +  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.returned                        | 4.0 KiB  
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B      
         
         
         -- Observe the total tracked tokens per-stream on n1. 2MiB is tracked for n1-n5;
         -- see last comment for an explanation why we're still deducting for n2, n3.
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
        --- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (26.69s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=17</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/057ffb5a-c86b-4210-becf-8d14bb954158?testReportRun=1&testReportShard=17&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-24.3 @ [e466862ab36572bdc894cec84536b45472bdeb5b](https://github.com/cockroachdb/cockroach/commits/e466862ab36572bdc894cec84536b45472bdeb5b):


```
        -  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.eval.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B     
        -  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B     
        +  kvflowcontrol.tokens.eval.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.eval.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
        +  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.returned                        | 8.0 KiB  
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B      
         
         
         -- Observe the total tracked tokens per-stream on n1. 2MiB is tracked for n1-n5;
         -- see last comment for an explanation why we're still deducting for n2, n3.
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
        --- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (27.95s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=17</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/e233fa58-4282-423c-8d0e-79f94d4b9164?testReportRun=1&testReportShard=18&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [472ea07a5232c98536293d13bb46cca59f9f2cd0](https://github.com/cockroachdb/cockroach/commits/472ea07a5232c98536293d13bb46cca59f9f2cd0):


```
        -  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.eval.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B     
        -  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B     
        +  kvflowcontrol.tokens.eval.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.eval.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
        +  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.returned                        | 4.0 KiB  
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B      
         
         
         -- Observe the total tracked tokens per-stream on n1. 2MiB is tracked for n1-n5;
         -- see last comment for an explanation why we're still deducting for n2, n3.
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
        --- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (26.91s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=18</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17389749?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17389749?buildTab=artifacts#/) on release-24.3 @ [b2d2353b876af1748607e155ecdfed9d4bba29d3](https://github.com/cockroachdb/cockroach/commits/b2d2353b876af1748607e155ecdfed9d4bba29d3):


```
        -  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.eval.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B     
        -  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B     
        +  kvflowcontrol.tokens.eval.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.eval.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
        +  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.returned                        | 8.0 KiB  
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B      
         
         
         -- Observe the total tracked tokens per-stream on n1. 2MiB is tracked for n1-n5;
         -- see last comment for an explanation why we're still deducting for n2, n3.
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
        --- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (6.48s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/d678dd0f-ef1a-4f70-8cb1-c4af76914afa?testReportRun=1&testReportShard=17&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-24.3 @ [b2d2353b876af1748607e155ecdfed9d4bba29d3](https://github.com/cockroachdb/cockroach/commits/b2d2353b876af1748607e155ecdfed9d4bba29d3):


```
        -  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.eval.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B     
        -  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B     
        +  kvflowcontrol.tokens.eval.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.eval.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
        +  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.returned                        | 4.0 KiB  
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.regular.available                       | 70 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B      
         
         
         -- Observe the total tracked tokens per-stream on n1. 2MiB is tracked for n1-n5;
         -- see last comment for an explanation why we're still deducting for n2, n3.
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
        --- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (7.40s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=17</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/kv/kvserver/BUILD.bazel
+++ b/pkg/kv/kvserver/BUILD.bazel
@@ -421,6 +421,7 @@ go_test(
         "//pkg/kv/kvserver/kvflowcontrol/kvflowdispatch",
         "//pkg/kv/kvserver/kvflowcontrol/kvflowinspectpb",
         "//pkg/kv/kvserver/kvflowcontrol/node_rac2",
+        "//pkg/kv/kvserver/kvflowcontrol/rac2",
         "//pkg/kv/kvserver/kvflowcontrol/replica_rac2",
         "//pkg/kv/kvserver/kvserverbase",
         "//pkg/kv/kvserver/kvserverpb",

--- a/pkg/kv/kvserver/flow_control_integration_test.go
+++ b/pkg/kv/kvserver/flow_control_integration_test.go
@@ -22,6 +22,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvflowcontrol"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvflowcontrol/kvflowinspectpb"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvflowcontrol/rac2"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/server"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
@@ -2253,6 +2254,10 @@ func TestFlowControlBasicV2(t *testing.T) {
 				n1 := sqlutils.MakeSQLRunner(tc.ServerConn(0))
 
 				h.waitForConnectedStreams(ctx, desc.RangeID, 3, 0 /* serverIdx */)
+				// Reset the token metrics, since a send queue may have instantly
+				// formed when adding one of the replicas, before being quickly
+				// drained.
+				h.resetV2TokenMetrics(ctx)
 
 				h.comment(`-- Flow token metrics, before issuing the 1MiB replicated write.`)
 				h.query(n1, v2FlowTokensQueryStr)
@@ -2342,6 +2347,10 @@ func TestFlowControlRangeSplitMergeV2(t *testing.T) {
 			require.NoError(t, err)
 
 			h.waitForConnectedStreams(ctx, desc.RangeID, 3, 0 /* serverIdx */)
+			// Reset the token metrics, since a send queue may have instantly
+			// formed when adding one of the replicas, before being quickly
+			// drained.
+			h.resetV2TokenMetrics(ctx)
 			h.log("sending put request to pre-split range")
 			h.put(ctx, k, 1<<20 /* 1MiB */, testFlowModeToPri(mode))
 			h.log("sent put request to pre-split range")
@@ -2464,6 +2473,10 @@ func TestFlowControlBlockedAdmissionV2(t *testing.T) {
 			require.NoError(t, err)
 			h.enableVerboseRaftMsgLoggingForRange(desc.RangeID)
 			h.waitForConnectedStreams(ctx, desc.RangeID, 3, 0 /* serverIdx */)
+			// Reset the token metrics, since a send queue may have instantly
+			// formed when adding one of the replicas, before being quickly
+			// drained.
+			h.resetV2TokenMetrics(ctx)
 
 			h.comment(`-- (Issuing 5 1MiB, 3x replicated write that's not admitted.)`)
 			h.log("sending put requests")
@@ -2579,6 +2592,10 @@ func TestFlowControlAdmissionPostSplitMergeV2(t *testing.T) {
 			require.NoError(t, err)
 
 			h.waitForConnectedStreams(ctx, desc.RangeID, 3, 0 /* serverIdx */)
+			// Reset the token metrics, since a send queue may have instantly
+			// formed when adding one of the replicas, before being quickly
+			// drained.
+			h.resetV2TokenMetrics(ctx)
 
 			h.log("sending put request to pre-split range")
 			h.put(ctx, k, 1<<20 /* 1MiB */, testFlowModeToPri(mode))
@@ -2722,6 +2739,10 @@ func TestFlowControlCrashedNodeV2(t *testing.T) {
 			require.NoError(t, err)
 			tc.TransferRangeLeaseOrFatal(t, desc, tc.Target(0))
 			h.waitForConnectedStreams(ctx, desc.RangeID, 2, 0 /* serverIdx */)
+			// Reset the token metrics, since a send queue may have instantly
+			// formed when adding one of the replicas, before being quickly
+			// drained.
+			h.resetV2TokenMetrics(ctx)
 
 			h.comment(`-- (Issuing 5x1MiB, 2x replicated writes that are not admitted.)`)
 			h.log("sending put requests")
@@ -2870,6 +2891,10 @@ func TestFlowControlRaftSnapshotV2(t *testing.T) {
 			repl := store.LookupReplica(roachpb.RKey(k))
 			require.NotNil(t, repl)
 			h.waitForConnectedStreams(ctx, repl.RangeID, 5, 0 /* serverIdx */)
+			// Reset the token metrics, since a send queue may have instantly
+			// formed when adding one of the replicas, before being quickly
+			// drained.
+			h.resetV2TokenMetrics(ctx)
 
 			// Set up a key to replicate across the cluster. We're going to modify this
 			// key and truncate the raft logs from that command after killing one of the
@@ -3085,6 +3110,10 @@ func TestFlowControlRaftMembershipV2(t *testing.T) {
 			desc, err := tc.LookupRange(k)
 			require.NoError(t, err)
 			h.waitForConnectedStreams(ctx, desc.RangeID, 3, 0 /* serverIdx */)
+			// Reset the token metrics, since a send queue may have instantly
+			// formed when adding one of the replicas, before being quickly
+			// drained.
+			h.resetV2TokenMetrics(ctx)
 
 			h.comment(`-- (Issuing 1x1MiB, 3x replicated write that's not admitted.)`)
 			h.put(ctx, k, 1<<20 /* 1MiB */, testFlowModeToPri(mode))
@@ -3224,6 +3253,10 @@ func TestFlowControlRaftMembershipRemoveSelfV2(t *testing.T) {
 				// Make sure the lease is on n1 and that we're triply connected.
 				tc.TransferRangeLeaseOrFatal(t, desc, tc.Target(0))
 				h.waitForConnectedStreams(ctx, desc.RangeID, 3, 0 /* serverIdx */)
+				// Reset the token metrics, since a send queue may have instantly
+				// formed when adding one of the replicas, before being quickly
+				// drained.
+				h.resetV2TokenMetrics(ctx)
 
 				h.comment(`-- (Issuing 1x1MiB, 3x replicated write that's not admitted.)`)
 				h.put(ctx, k, 1<<20 /* 1MiB */, testFlowModeToPri(mode))
@@ -3353,6 +3386,10 @@ func TestFlowControlClassPrioritizationV2(t *testing.T) {
 			desc, err := tc.LookupRange(k)
 			require.NoError(t, err)
 			h.waitForConnectedStreams(ctx, desc.RangeID, 3, 0 /* serverIdx */)
+			// Reset the token metrics, since a send queue may have instantly
+			// formed when adding one of the replicas, before being quickly
+			// drained.
+			h.resetV2TokenMetrics(ctx)
 
 			h.comment(`-- (Issuing 1x1MiB, 3x replicated elastic write that's not admitted.)`)
 			h.put(ctx, k, 1<<20 /* 1MiB */, testFlowModeToPri(mode))
@@ -3469,6 +3506,10 @@ func TestFlowControlUnquiescedRangeV2(t *testing.T) {
 			n1 := sqlutils.MakeSQLRunner(tc.ServerConn(0))
 
 			h.waitForConnectedStreams(ctx, desc.RangeID, 3, 0 /* serverIdx */)
+			// Reset the token metrics, since a send queue may have instantly
+			// formed when adding one of the replicas, before being quickly
+			// drained.
+			h.resetV2TokenMetrics(ctx)
 
 			h.comment(`-- (Issuing 1x1MiB, 3x replicated elastic write that's not admitted.)`)
 			h.put(ctx, k, 1<<20 /* 1MiB */, admissionpb.BulkNormalPri)
@@ -3571,6 +3612,10 @@ func TestFlowControlTransferLeaseV2(t *testing.T) {
 			desc, err := tc.LookupRange(k)
 			require.NoError(t, err)
 			h.waitForConnectedStreams(ctx, desc.RangeID, 3, 0 /* serverIdx */)
+			// Reset the token metrics, since a send queue may have instantly
+			// formed when adding one of the replicas, before being quickly
+			// drained.
+			h.resetV2TokenMetrics(ctx)
 
 			h.comment(`-- (Issuing 1x1MiB, 3x replicated write that's not admitted.)`)
 			h.put(ctx, k, 1<<20 /* 1MiB */, testFlowModeToPri(mode))
@@ -3664,6 +3709,10 @@ func TestFlowControlLeaderNotLeaseholderV2(t *testing.T) {
 			desc, err := tc.LookupRange(k)
 			require.NoError(t, err)
 			h.waitForConnectedStreams(ctx, desc.RangeID, 3, 0 /* serverIdx */)
+			// Reset the token metrics, since a send queue may have instantly
+			// formed when adding one of the replicas, before being quickly
+			// drained.
+			h.resetV2TokenMetrics(ctx)
 
 			h.comment(`-- (Issuing 1x1MiB, 3x replicated write that's not admitted.)`)
 			h.put(ctx, k, 1<<20 /* 1MiB */, testFlowModeToPri(mode))
@@ -3780,6 +3829,10 @@ func TestFlowControlGranterAdmitOneByOneV2(t *testing.T) {
 			desc, err := tc.LookupRange(k)
 			require.NoError(t, err)
 			h.waitForConnectedStreams(ctx, desc.RangeID, 3, 0 /* serverIdx */)
+			// Reset the token metrics, since a send queue may have instantly
+			// formed when adding one of the replicas, before being quickly
+			// drained.
+			h.resetV2TokenMetrics(ctx)
 
 			h.comment(`-- (Issuing 1024*1KiB, 3x replicated writes that are not admitted.)`)
 			h.log("sending put requests")
@@ -4862,6 +4915,18 @@ func (h *flowControlTestHelper) enableVerboseRaftMsgLoggingForRange(rangeID roac
 				IncomingRaftMessageHandler: si,
 				unreliableRaftHandlerFuncs: noopRaftHandlerFuncs(),
 			})
+	}
+}
+
+func (h *flowControlTestHelper) resetV2TokenMetrics(ctx context.Context) {
+	for _, server := range h.tc.Servers {
+		require.NoError(h.t, server.GetStores().(*kvserver.Stores).VisitStores(func(s *kvserver.Store) error {
+			s.GetStoreConfig().KVFlowStreamTokenProvider.Metrics().(*rac2.TokenMetrics).TestingClear()
+			_, err := s.ComputeMetricsPeriodically(ctx, nil, 0)
+			require.NoError(h.t, err)
+			s.GetStoreConfig().KVFlowStreamTokenProvider.UpdateMetricGauges()
+			return nil
+		}))
 	}
 }
 

--- a/pkg/kv/kvserver/kvflowcontrol/rac2/metrics.go
+++ b/pkg/kv/kvserver/kvflowcontrol/rac2/metrics.go
@@ -197,6 +197,34 @@ func NewTokenMetrics() *TokenMetrics {
 	return m
 }
 
+// TestingClear is used in tests to reset the metrics.
+func (m *TokenMetrics) TestingClear() {
+	// NB: we only clear the counter metrics, as the stream metrics are gauges.
+	for _, typ := range []TokenType{
+		EvalToken,
+		SendToken,
+	} {
+		for _, wc := range []admissionpb.WorkClass{
+			admissionpb.RegularWorkClass,
+			admissionpb.ElasticWorkClass,
+		} {
+			m.CounterMetrics[typ].Deducted[wc].Clear()
+			m.CounterMetrics[typ].Returned[wc].Clear()
+			m.CounterMetrics[typ].Unaccounted[wc].Clear()
+			m.CounterMetrics[typ].Disconnected[wc].Clear()
+			if typ == SendToken {
+				m.CounterMetrics[typ].SendQueue[0].ForceFlushDeducted.Clear()
+				for _, wc := range []admissionpb.WorkClass{
+					admissionpb.RegularWorkClass,
+					admissionpb.ElasticWorkClass,
+				} {
+					m.CounterMetrics[typ].SendQueue[0].PreventionDeducted[wc].Clear()
+				}
+			}
+		}
+	}
+}
+
 type TokenCounterMetrics struct {
 	Deducted     [admissionpb.NumWorkClasses]*metric.Counter
 	Returned     [admissionpb.NumWorkClasses]*metric.Counter

